From 8dd86c560119f3ec0622a735251ca7fa6cbcd3bb Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Mon, 11 Aug 2014 10:09:02 -0400
Subject: [net] clear local_df when passing skb between namespaces

Message-id: <1407751742-7303-1-git-send-email-jpirko@redhat.com>
Patchwork-id: 87593
O-Subject: [patch rhel7 net] net: clear local_df when passing skb between namespaces
Bugzilla: 1128223
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Jesper Brouer <jbrouer@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>

BZ1128223
https://brewweb.devel.redhat.com/taskinfo?taskID=7827645

upstream commit 239c78db9c41a8f524cce60507440d72229d73bc
Author: Hannes Frederic Sowa <hannes@stressinduktion.org>
Date:   Thu Dec 5 23:29:19 2013 +0100

    net: clear local_df when passing skb between namespaces

    We must clear local_df when passing the skb between namespaces as the
    packet is not local to the new namespace any more and thus may not get
    fragmented by local rules. Fred Templin noticed that other namespaces
    do fragment IPv6 packets while forwarding. Instead they should have send
    back a PTB.

    The same problem should be present when forwarding DF-IPv4 packets
    between namespaces.

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 7f65ea9..792d691 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -3643,6 +3643,7 @@ void skb_scrub_packet(struct sk_buff *skb)
  skb->tstamp.tv64 = 0;
  skb->pkt_type = PACKET_HOST;
  skb->skb_iif = 0;
+ skb->local_df = 0;
  skb_dst_drop(skb);
  skb->mark = 0;
  secpath_reset(skb);
-- 
1.7.1